lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2966,6 +2970,33 @@ void IRGenDebugInfoImpl::emitImport(ImportDecl *D) {
   ImportedModules.insert(Imported.importedModule);
 }
 
+/// This is effectively \p clang::CGDebugInfo::getCallSiteRelatedAttrs().
+llvm::DINode::DIFlags IRGenDebugInfoImpl::getCallSiteRelatedAttrs() const {
+  auto Optimize = IGM.getClangASTContext().getLangOpts().Optimize;
+  auto Loader = IGM.getSILModule().getASTContext().getClangModuleLoader();
+  auto *Importer = static_cast<ClangImporter *>(&*Loader);
+  auto &CGO = Importer->getCodeGenOpts();
+
+  // Call site-related attributes are only useful in optimized programs, and
+  // when there's a possibility of debugging backtraces.
+  if (!Optimize || CGO.getDebugInfo() == llvm::codegenoptions::NoDebugInfo ||
+      CGO.getDebugInfo() == llvm::codegenoptions::LocTrackingOnly)
+    return llvm::DINode::FlagZero;
+
+  // Call site-related attributes are available in DWARF v5. Some debuggers,
+  // while not fully DWARF v5-compliant, may accept these attributes as if they
+  // were part of DWARF v4.
+  bool SupportsDWARFv4Ext =
+      CGO.DwarfVersion == 4 &&
+      (CGO.getDebuggerTuning() == llvm::DebuggerKind::LLDB ||
+       CGO.getDebuggerTuning() == llvm::DebuggerKind::GDB);
```

Comment:
This the embedded Clang's flag, we should only check Swift LangOpts here. Generally, GDB debugger tuning+swift isn't a meaningful combination.